### DISCUSSION TITLE
Best practice for generating typescript defs from queries

### DISCUSSION THREAD

**Discussion by moverholt** [vercel/swr#1138] [Q&A]
Hello!

Does SWR have a best practice for generating typescript def files? I want to type my queries using generated types something like:
`const { data } = useSWR<GENERATED_TYPE_RESP, GENERATED_TYPE_VARS>(...)`

With Google, I found the graphql code gen SWR plugin. However, I wanted to ask someone with SWR experience their thoughts for generating types?

Thank you!
-matt
(Working on my first project with SWR, in the past i've worked on projects with Apollo & Relay)

**Comment by anothertempore** [comment]
Not very automated, but you can try this: https://transform.tools/json-to-typescript
